File: Scripts/De-activate/ByLanguage/C#/DE-ACTIVATE.cs
// Start of script
// A script to de-activate Ransomware if the target server is no longer available
// Script is not yet functional
using System.Net.NetworkInformation;    
// Taken from https://stackoverflow.com/a/11804416 | slightly modified
public static bool PingHost(string nameOrAddress)
{
    bool pingable = false;
    Ping pinger = null;

    try
    {
        pinger = new Ping();
        PingReply reply = pinger.Send(nameOrAddress);
        pingable = reply.Status == IPStatus.Success;
    }
    catch (PingException)
    {
        // Discard PingExceptions and return false;
    }
    finally
    {
        if (pinger != null)
        {
            pinger.Dispose();
        }
    }

    return pingable;
    if pingable {
      return 0
    } else {
      return deactivate;
    }
}
public static int deactivate(String, str)
{
  Console.WriteLine("De-activating Ransomware, as server could not be reached");
  /* Script needs to go between here */
  // Scipt goes here
  /* Script needs to go between here */
}
public static void main(String, str)
{
  return PingHost;
}
// File info
// File type: C# source file (*.cs *.csx)
// File version: 1 (2022, Monday, August 22nd at 5:05 pm PST)
// Line count (including blank lines and compiler line): 52
// End of script

File: Scripts/De-activate/ByLanguage/C++/DE-ACTIVATE.cxx
// Start of script
// A script to de-activate Ransomware if the target server is no longer available
// Script is not yet functional, and was modified from a C# example with a library that may not exist in C++
import *;
void PingHost() {
    pingable = bool(false);
    Ping pinger = "null";
    try {
        pinger = new Ping();
        PingReply.reply = pinger.Send(nameOrAddress);
        pingable = reply.Status == IPStatus.Success;
        break;
    }
    catch (PingException) {
        // Discard PingExceptions and return false;
    }
    final {
        if (pinger != null) {
            pinger.Dispose();
            break;
    }}
    return pingable();
    if pingable = true {
      return 0;
      break;
    } else {
      return deactivate();
      break;
}}
void deactivate() {
  std::cout << "De-activating Ransomware, as server could not be reached";
  /* Script needs to go between here */
  // Scipt goes here
  /* Script needs to go between here */
}
void main() {
  return PingHost();
  break;
}
return main();
break;
// File info
// File type: C++ source file (*.cpp *.cxx *.hh *.C *.hxx *.h++ *.c++)
// File version: 1 (2022, Monday, August 22nd at 5:26 pm PST)
// Line count (including blank lines and compiler line): 47
// End of script
